### DISCUSSION TITLE
Error: Module not found: Can't resolve 'fs'

### DISCUSSION THREAD

**Discussion by AstridOttmann** [neuefische/questions#515] [Web] [answered]
I want to use the deepl-node API for translation. 
I installed the npm package, got an authentification key and a translator function.  
` const translator = new deepl.Translator(authKey);

  (async () => {
    const result = await translator.translateText("Hello, world!", null, "fr");
    console.log(result.text);   // Bonjour, le monde ! The console-log works!
  })();`

Now I get this error: 
Module not found: Can't resolve 'fs'

I found the hint, that I have to make a change in my next.config, but for that I find different solutions. 
I tried this: 
`module.exports = {
  webpack: (config, { isServer }) => {
    // Fixes npm packages that depend on `fs` module
    if (!isServer) {
      config.node = {
        fs: 'empty'
      }
    }

    return config
  }
}`

But it didn't work  respectively I don't know how to integrate it in my config properly.

**Comment by pixelass** [comment, marked as answer]
You have to use this in your serverless function (e.g, `/api/deepl.js`) and then communicate with your api route with REST.

Also please provide more context. Where are you using this? Can you give us a link to the actual implementation?

An adjustment in `next.config.js` is not required and should not be made for this case.

**Reply by AstridOttmann** [reply]
I tried in in a local test-project, so I can't provide a link to the repro 
At the moment I can't go on because of the error. Do I have to deinstall the npm package (respectively delete it from the package.json) to get rid of the error. 

I don't understand what you mean by here : "You have to use this in your serverless function (e.g, /api/deepl.js) and then communicate with your api route with REST."
So far I have no api-routes, only working with localStorage.

**Reply by pixelass** [reply]
nextjs has a special folder called `api` which is nested in `pages`.

This allows you to run Node.js on the server and communicate with the client. 

Since the library requires Node.js (as the name suggests) you cannot use this package on the client (everything that is not in `pages/api/*`)

Here's a link to the docs: https://nextjs.org/docs/api-routes/introduction

**Reply by pixelass** [reply]
Maybe this helps:

![Untitled-2022-09-09-1018](https://user-images.githubusercontent.com/1148334/217507012-7ef3b71e-88b7-425b-bd38-c4e8e89bfc3f.png)

**Reply by pixelass** [reply]
If you do not want to use api routes, then you cannot use packages that require Node.js.

**Reply by AstridOttmann** [reply]
Sorry, I don't get it. 
We used the api only with mongo.

**Reply by RobinWitt** [reply]
Think about the DeepL API like it is a MongoDB. You want to communicate with it through the API methods we used in the course.

- your component/page sends a request to your API, for example `/pages/api/translate/index.js` (this is what you need to create)
- the `request body` should be the word you want to translate
- the API file gets the "word" as `request` from your component and adds the `header` with your authentication, and sends it to the DeepL API
- the `response` from the DeepL API is sent to your API route and then sent to the component

**Reply by AstridOttmann** [reply]
Thank you, Robin.
